proctor-common/src/main/java/com/indeed/proctor/common/Proctor.java

```diff
@@ -522,23 +523,53 @@ private boolean isIncognitoEnabled(@Nonnull final Map<String, Object> inputConte
                 .orElse(false);
     }
 
-    private Map<String, PayloadProperty> getProperties(
-            final String testName, final TestBucket testBucket) {
+    private void populateProperties(
+            final String testName,
+            final TestBucket testBucket,
+            final Map<String, PayloadProperty> testProperties) {
         final Payload payload = testBucket.getPayload();
         if (payload != null && payload.getJson() != null) {
-            final SortedMap<String, PayloadProperty> testProperties = new TreeMap<>();
             payload.getJson()
                     .fields()
                     .forEachRemaining(
-                            field ->
+                            field -> {
+                                final PayloadProperty curr = testProperties.get(field.getKey());
+                                if (curr == null) {
                                     testProperties.put(
                                             field.getKey(),
                                             PayloadProperty.builder()
                                                     .value(field.getValue())
                                                     .testName(testName)
-                                                    .build()));
-            return testProperties;
+                                                    .build());
+                                } else {
+                                    final PayloadExperimentConfig currPayloadConfig =
+                                            testChoosers
+                                                    .get(curr.getTestName())
+                                                    .getTestDefinition()
+                                                    .getPayloadExperimentConfig();
+                                    final PayloadExperimentConfig newPayloadConfig =
+                                            testChoosers
+                                                    .get(testName)
+                                                    .getTestDefinition()
+                                                    .getPayloadExperimentConfig();
+                                    if (currPayloadConfig != null
```

Comment:
this part makes sense. may be easier to read if extracting to a method such as isLowerPriority(currPayloadConfig, newPayloadConfig)

Reply:
Created helper method for this